File: typescript/packages/x402/src/signers/agentWalletProvider.ts
/**
 * Helpers for constructing agent-wallet providers without direct wallet imports.
 */

export interface TronProviderOptions {
  fullNode?: string;
  solidityNode?: string;
  eventServer?: string;
  privateKey?: string;
  apiKey?: string;
  keystore?: {
    path?: string;
    password?: string;
  };
}

export interface FlashProviderOptions extends TronProviderOptions {
  flashNode?: string;
  privyAppId?: string;
  privyAppSecret?: string;
  walletId?: string;
}

export async function createTronProvider(options: TronProviderOptions = {}): Promise<any> {
  // @ts-ignore agent-wallet is an optional dependency
  const { TronProvider } = await import('agent-wallet/wallet');
  return new TronProvider({
    ...options,
    keystore: options.keystore,
  });
}

export async function createFlashProvider(options: FlashProviderOptions = {}): Promise<any> {
  // @ts-ignore agent-wallet is an optional dependency
  const { FlashProvider } = await import('agent-wallet/wallet');
  const provider = new FlashProvider({
    ...options,
    keystore: options.keystore,
  });
  await provider.init();
  return provider;
}

File: python/x402/src/x402_tron/signers/agent_wallet_provider.py
"""Helpers for constructing agent-wallet providers without direct wallet imports."""

from __future__ import annotations

from typing import Any


async def create_tron_provider(
    *,
    keystore_path: str | None = None,
) -> Any:
    """Create an agent-wallet TronProvider via async ``create()`` factory."""
    from wallet import TronProvider

    return await TronProvider.create(
        keystore_path=keystore_path,
    )
